### DISCUSSION TITLE
TypeError: beanie.odm.queries.find.FindOne.update() got multiple values for keyword argument 'response_type'

### DISCUSSION THREAD

**Discussion by DrOncogene** [BeanieODM/beanie#779] [Question]
I was getting `TypeError: beanie.odm.queries.find.FindOne.update() got multiple values for keyword argument 'response_type'` while providing response_type argument to the update method. The code snippet is below

```
room = await Room.find_one(
        Room.id == PydanticObjectId(room_id),
        fetch_links=True,
)

there are other things being done in between, mainly some validation prior to the update

room = await room.update(
        AddToSet({Room.members: {"$each": members_ref}}),
        response_type=UpdateResponse.NEW_DOCUMENT
)
```

But it was stated in the docs [here](https://beanie-odm.dev/tutorial/updating-%26-deleting/#response-type) that the `.update()` and `.upsert()` methods can take `response_type` arguments.  

![image](https://github.com/roman-right/beanie/assets/66273502/95854c5b-ee07-4e17-8a84-5c280375fcf4)

How do I solve this?

**Comment by hgalytoby** [comment]
I haven't used it but I found an example from tests that you can look at.

https://github.com/roman-right/beanie/blob/main/tests/odm/query/test_update.py#L198

**Comment by MatthieuRochetteAO** [comment]
Hi, I get the same error when trying to use `.update()` on my Document, but it works when using `.upsert()`.

Looking at the [source code of the update method](https://github.com/roman-right/beanie/blob/main/beanie/odm/documents.py#L710), the `response_type` is set to `UpdateResponse.NEW_DOCUMENT` forcefully, so in our cases we can just skip the parameter and it should behave as we want it to.

But indeed, the documentation should be updated to reflect this, or the code to allow the `response_type` to be overriden in the `.update()` parameters.

**Comment by Jain2098** [comment]
+1 
Not working on methods like: `.save()` / `.update() `
